File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-7.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int j=nums.size()-1,i=0;
        while(i<=j){
           int  m=i+(j-i)/2;

            if(nums[m]==target)return m;
            else if(nums[j]==target)return j;
            else if(nums[i]==target)return i;


            else if(nums[m]<target){
                if(nums[j]>target or nums[m]>nums[j]) i=m+1;
                else j=m-1;
            }
            else {
                if(nums[i]<target or nums[m]<nums[i])j=m-1;
                else i=m+1;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-7.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        x=False
        z=False
        y=False
        for [a,b,c] in triplets:
            x= x | (a==target[0] and b<=target[1] and c<=target[2])  
            y= y | (a<=target[0] and b==target[1] and c<=target[2])  
            z= z | (a<=target[0] and b<=target[1] and c==target[2])  
            if z and x and y :
                return True
        return False